sc-memory/sc-core/src/sc_memory_ext.c

```diff
@@ -63,52 +63,36 @@ sc_char * _sc_module_build_path(sc_char const * directory, sc_char const * modul
   return g_strconcat(directory, "/", module_name, null_ptr);
 }
 
-sc_result sc_ext_initialize(
-    sc_char const * ext_dir_path,
-    sc_char const ** enabled_list,
-    sc_addr const init_memory_generated_structure)
+sc_result _sc_ext_collect_extensions_from_directory(sc_char const * extension_directory_path)
 {
-  sc_unused(&enabled_list);
-
-  GDir * ext_dir = null_ptr;
+  GDir * extension_directory = null_ptr;
   sc_char const * file_name = null_ptr;
   fModuleInitializeFunc initialize_func = null_ptr;
   fModuleInitializeFunc shutdown_func = null_ptr;
 
-  // doesn't need to initialize extensions
-  if (ext_dir_path == null_ptr)
-    return SC_RESULT_OK;
-
-  // check if modules supported on this platform
-  if (g_module_supported() == SC_FALSE)
-  {
-    sc_warning("Modules not supported on this platform");
-    return SC_RESULT_ERROR;
-  }
-
-  sc_message("Initialize extensions from %s", ext_dir_path);
+  sc_message("Collect extensions from %s", extension_directory_path);
```

Comment:
if sc-machine.ini has one directory repeated multiple times then extensions from it will be loaded multiple times

extensions = ./bin/extensions;./bin/../bin/extensions;./bin/../bin/../bin/extensions;./bin/../bin/../bin/../bin/../bin/extensions
![image](https://github.com/user-attachments/assets/e020d6b0-a4eb-461a-b77b-4760d73eee2d)

Reply:
![image](https://github.com/user-attachments/assets/73fd587d-d446-406b-83cd-03c6e152d02a)